representatives_votes/contrib/parltrack/import_votes.py

```diff
@@ -1,185 +1,23 @@
-
-        for receiver, response in responses:
-            if response is False:
-                logger.debug(
-                    'Skipping dossier %s', proposal_data.get(
-                        'epref', proposal_data['title']))
-                return
-
-        positions = ['For', 'Abstain', 'Against']
-        logger.info(
-            'Looking for votes in proposal {}'.format(proposal_display))
-        for position in positions:
-            for group_vote_data in proposal_data.get(
-                    position,
-                    {}).get(
-                    'groups',
-                    {}):
-                for vote_data in group_vote_data['votes']:
-                    if not isinstance(vote_data, dict):
-                        logger.error('Skipping vote data %s for proposal %s',
-                                     vote_data, proposal_data['_id'])
-                        continue
-
-                    representative_pk = self.get_representative(vote_data)
-
-                    if representative_pk is None:
-                        logger.error('Could not find mep for %s', vote_data)
-                        continue
-
-                    representative_name = vote_data.get('orig', '')
-
-                    changed = False
-                    try:
-                        vote = Vote.objects.get(
-                            representative_id=representative_pk,
-                            proposal_id=proposal.pk)
-                    except Vote.DoesNotExist:
-                        vote = Vote(proposal_id=proposal.pk,
-                            representative_id=representative_pk)
-                        changed = True
-
-                    if vote.position != position.lower():
-                        changed = True
-                        vote.position = position.lower()
-
-                    if vote.representative_name != representative_name:
-                        changed = True
-                        vote.representative_name = representative_name
-
-                    if changed:
-                        vote.save()
-                        logger.debug('Save vote %s for MEP %s on %s #%s to %s',
-                            vote.pk, representative_pk, proposal_data['title'],
-                            proposal.pk, position)
-
-        return proposal
-
-    def index_dossiers(self):
-        self.cache['dossiers'] = {
-            d[0]: d[1] for d in Dossier.objects.values_list('reference', 'pk')
-        }
-
-    def get_dossier(self, reference):
-        return self.cache['dossiers'].get(reference, None)
-
-    def index_representatives(self):
-        self.cache['meps'] = {int(l[0]): l[1] for l in
-            Representative.objects.values_list('remote_id', 'pk')}
-
-    def get_representative(self, vote_data):
-        if vote_data.get('ep_id', None) is None:
-            return
-        return self.cache['meps'].get(int(vote_data['ep_id']), None)
+vote_pre_import = vote_signal
```

Comment:
This signal existed before, just with a different name.

`vote_signal` happens when the previously existing `vote_pre_import` signal happened; If I get it right, it runs after updating vote counts in a proposal and before updating votes. It also enables receivers to cancel the update.

Since I moved actual updating code from import_*.py scripts to a common lib.py code, I had to move the signal as well. But this `vote_pre_import` signal was exported and used outside, so that's what I had to do to keep it the same way.